designs/event-streams.md

```diff
@@ -0,0 +1,275 @@
+# Event Streams
+
+Event streams represent a behavioral difference in Smithy operations. Most
+operations work philosophically like functions in python - you provide some
+parameters once, and get results once. Event streams, on the other hand,
+represent a continual exchange of data which may be flow in one direction
+or in both directions (a.k.a. a "bidirectional" or "duplex" stream).
+
+To facilitate these different usage scenarios, the return type event stream
+operations are altered to provide customers with persistent stream objects
+that they can write or read to.
+
```

Comment:
Might help to show customer usage example before diving into the components.

Reply:
I personally prefer examples to be at the end, but maybe I'm misunderstanding what you're asking for.